### PR TITLE
[WIP] Optimize release history generation and timestamps

### PR DESCRIPTION
**Pull request by daniel-beck**
There are three related optimizations in this PR:

1. Only parse the plugin POM on demand rather than `Plugin` creation.
1. Use the artifact modification date rather than the `MANIFEST.MF` change time inside the `hpi`.
1. Do not include `title` or `wiki` in `release-history.json` entries older than a month

The **first change** should be fairly straightforward and comes with no change in behavior, it just enables the third one.

The **second change** will change almost all timestamps due to a time zone dependency I haven't figured out. Most release times change by 0-14 hours ± 5 minutes. Some, e.g. Surround SCM, change by weeks, something I don't understand and asked the maintainer at https://github.com/jenkinsci/Surround-SCM-plugin/commit/90a4e49f927e9444dceb4eaf54b897a2c414e7c7#commitcomment-21274381 -- basically, the release commit predates the upload date by 1-2 weeks here.

Only few current plugin releases (i.e. entries in update-center.json) are affected to that magnitude, and it's probably better to use a globally consistent reference date for plugins anyway.

One side effect if this is that the migration to Artifactory in 2010 will result in *many* older plugins having their release date set to April/May 2010 even if they were released earlier. Seems like an acceptable trade-off to me.

Finally, the **third change** is a straightforward optimization enabled by the first two: The update center generator needs to open the POM for the wiki URL and title, so skip that for old releases.

The only currently known consumer of this file is https://github.com/jenkins-infra/jenkins.io/blob/master/scripts/release.rss.groovy which ignores entries older than 30 days.

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/update_center/MavenArtifact.java [hunk 1]
```diff
@@ -165,27 +166,26 @@ public String getTimestampAsString() throws IOException {
...
-        Date tsDate;
         
-        try {
-            tsDate = bdf.parse(bdf.format(new Date(lastModified)));
-        } catch (ParseException pe) {
-            throw new IOException(pe.getMessage());
-        }
 
-        return tsDate;
+        Date lastModifiedDate = new Date(lastModified);
```
Comment (daniel-beck) [inline src/main/java/org/jvnet/hudson/update_center/MavenArtifact.java]: Probably unnecessary.

File: src/main/java/org/jvnet/hudson/update_center/MavenArtifact.java [hunk 2]
```diff
@@ -165,27 +166,26 @@ public String getTimestampAsString() throws IOException {
...
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        return cal.getTime();
     }
     
     public static SimpleDateFormat getDateFormat() {
-        return new SimpleDateFormat("MMM dd, yyyy", Locale.US);
+        SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy", Locale.US);
+        sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
```
Comment (daniel-beck) [inline src/main/java/org/jvnet/hudson/update_center/MavenArtifact.java]: Time stamps appear to be one hour off from what's shown on Artifactory and I think this was intended to fix that, but didn't work out.

**Comment by orrc** [comment]
Oops, thought this was still WIP. Anyway, 👍

**Comment by daniel-beck** [comment]
@orrc Thanks. I forgot to remove the marker, it was never really WIP -- Just wanted to prevent premature merging. I checked the changelog contents and the last 30 days seem fine, some dates (and corresponding order) changed by up to a day, but otherwise same as before.

**Pull request merged**